jekyll/_cci2/optimization-cookbook.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: classic-docs
+title: "CircleCI Optimizations Cookbook"
+short-title: "Optimizations Cookbook"
+description: "Starting point for Optimizations Cookbook"
+categories: [getting-started]
+order: 1
+---
+
+The *CircleCI Optimizations Cookbook* is a collection of individual use cases (referred to as "recipes") that provide you with detailed, step-by-step instructions on how to perform various optimization tasks using CircleCI resources. This guide, and it associated sections, will enable you to quickly and easily perform easily repeatable optimization tasks on the CircleCI platform.
+
+* TOC
+{:toc}
+
+## Introduction
+
+When work with the CircleCI platform, one of the most critical tasks you will want to complete is to optimize your jobs, builds, workflows, and other tasks to prevent performance degradation, latency, and concurrency. Optimizing your tasks is not simply a function of lessening the workload or number of tasks, but rather, reviewing your optimization and caching strategies to ensure you are meeting your business and technical requirements while also maintaining stability and reliability.
+
+CircleCI has developed several different approaches you may wish to use to better optimize these tasks while also leveraging the power of the CircleCI platform and its associated resources (e.g. orbs). The sections below describe various strategies and methodologies you can use to optimize your organization tasks.
+
+### Optimization Recipes
+
+The table below lists the different optimization tasks that can be performed using the CircleCI platform.
+
+Recipe | Description 
+------------|-----------
+Single-Threading Builds (Queueing) | This section describes how you can use a specific single-threading (queueing) orb to ensure proper job and build completion without concurrency (queueing subsequent builds or jobs).
+Caching |  This section describes how you can use different caching strategies in your implemenations to optimize your builds and workflows.
+Workflow Optimizations | This section describes how you can optimize your workflows.
+
+## Single-Threading Builds (Queuing)
+
+One of the most common tasks you may encounter when using the CircleCI platform for builds and jobs is managing multiple jobs and builds simultaneously to ensure your operations do not fail because of system timeouts. This becomes especially important when you have multiple contributors and committers working in the same environment. Because the CircleCI platform was designed to handle multiple tasks simultaneously without performance degradation or latency, concurrency may sometimes become an issue if there are a large number of jobs being queued, waiting for a previous job to be completed before the new job can be initiated, and the system timeout is set too low. In this case, one job will be completed, and other jobs may fail due to this timeout setting.
```

Comment:
I feel like we need to be careful using the terms job and build like this - my understanding 'build' is kind of a generic terms that has many subtle meanings inside and outside circleci, whereas job is a very specific term, e.g. you can have a build job. 

Here, for example, could just say: "One of the most common tasks you may encounter when using the CircleCI platform is managing multiple jobs simultaneously to ensure your operations do not fail because of system timeouts." and this has the same meaning but is less ambiguous?

That being said - I'm struggling to understand this use case - why would we want to prevent a job from running concurrently with others?

Reply:
Is system timeout something the customer has control over?

Reply:
Resolved in document. Thanks for the great comments @rosieyohannan 👍